File: source_funcs/go_transcoder/CHECK_IF_ALL_THE_ELEMENTS_CAN_BE_MADE_OF_SAME_PARITY_BY_INVERTING_ADJACENT_ELEMENTS/CHECK_IF_ALL_THE_ELEMENTS_CAN_BE_MADE_OF_SAME_PARITY_BY_INVERTING_ADJACENT_ELEMENTS.go
package main

func f_gold(a []int, n int) bool {
	var (
		count_odd  int = 0
		count_even int = 0
	)
	for i := int(0); i < n; i++ {
		if a[i]&1 != 0 {
			count_odd++
		} else {
			count_even++
		}
	}
	if count_odd%2 != 0 && count_even%2 != 0 {
		return false
	} else {
		return true
	}
}

File: rust_funcs/cpp_transcoder/CHECK_IF_ALL_THE_ELEMENTS_CAN_BE_MADE_OF_SAME_PARITY_BY_INVERTING_ADJACENT_ELEMENTS/CHECK_IF_ALL_THE_ELEMENTS_CAN_BE_MADE_OF_SAME_PARITY_BY_INVERTING_ADJACENT_ELEMENTS.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::fs::File;
use std::io::{BufRead, BufReader};

fn f_gold(a: [i32;2], n: i32) -> i32 {
    let mut count_odd = 0;
    let mut count_even = 0;
    for i in 0..n {
        if a[i as usize] & 1 == 1 {
            count_odd += 1; 
        } else {
            count_even += 1;
        }
    }
    if count_odd % 2 == 1 && count_even % 2 == 1 {
        0
    } else {
        1
    } 
}
